java/src/jmri/jmrit/logixng/expressions/swing/ConnectionNameSwing.java

```diff
@@ -34,110 +38,112 @@ public ConnectionNameSwing(JDialog dialog) {
 
     @Override
     protected void createPanel(@CheckForNull Base object, @Nonnull JPanel buttonPanel) {
-        ConnectionName action = (ConnectionName) object;
-        if (action == null) {
-            // Create a temporary action
-            action = new ConnectionName("IQDE1", null);
-        }
+        // Create a temporary action if object is null
+        final ConnectionName action = (object != null)
+                ? (ConnectionName) object
+                : new ConnectionName("IQDE1", null);
 
-        _selectConnectionNameSwing = new LogixNG_SelectStringSwing(getJDialog(), this);
+        String[] manufactureNameList = manager.getConnectionManufacturers();
+        _connectionConfigs = manager.getConnections();
 
         panel = new JPanel();
-        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
 
-        JPanel tabbedPaneTopic = _selectConnectionNameSwing.createPanel(action.getSelectConnectionName());
-        panel.add(tabbedPaneTopic);
+        String selectedManufacturer = action.getManufacturer();
+        if (selectedManufacturer == null
+                || selectedManufacturer.isBlank()
+                || NONE_SELECTED.equals(selectedManufacturer)) {
+            if (_connectionConfigs.length > 0) {
+                selectedManufacturer = _connectionConfigs[0].getManufacturer();
+            }
+        }
 
+        JPanel manufacturerPanel = new JPanel();
+        manufacturerPanel.setLayout(new BoxLayout(manufacturerPanel, BoxLayout.Y_AXIS));
+        manufacturerPanel.add(new JLabel("System manufacturer"));
+        _manufacturerComboBox = new JComboBox<>();
```

Comment:
The combo boxes need JComboBoxUtil.setupComboBoxMaxRows​()

Reply:
Thanks. Done.
Do I need to do this every time the combo box is cleared and filled with new content?

Reply:
No.  Only after creating a new instance.  It changes the maximum row count from the default value of 8 to the screen capacity or the user preference limit.